### PR TITLE
bAdd brainflow dev

### PR DESCRIPTION
**Pull request by romanov6416**

### CODE REVIEW

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#include "BrainFlowNodes.h"
+
+#include "../../EngineManager.h"
+
+namespace
+{
+	bool operator==(const BrainFlowInputParams& left, const BrainFlowInputParams& right)
```
Comment (Andrey1994) [inline src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:7]: I think BrainFlowInputParams has operator less implemented, but there is no == operator, we can add it there or you can use less here to implement eq operator

Comment (Andrey1994) [reply src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:7]: But in BrainFlowNode more likely need to check board id also

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			&& left.file == right.file;
+	}
+}
+
+void BrainFlowNode::Init()
+{
+	DeviceInputNode::Init();
+	{
+		Core::AttributeSettings* attribute = RegisterAttribute("Board ID", "boardID", "Identificator of the board", Core::ATTRIBUTE_INTERFACETYPE_INTSPINNER);
+		attribute->SetDefaultValue(Core::AttributeInt32::Create(-1));
```
Comment (Andrey1994) [inline src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:26]: replace hardcode by values from enum

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		attribute->SetDefaultValue(Core::AttributeString::Create(""));
+	}
+	{
+		Core::AttributeSettings* attribute = RegisterAttribute("IP address", "ipAddress", "IP address", Core::ATTRIBUTE_INTERFACETYPE_STRING);
+		attribute->SetDefaultValue(Core::AttributeString::Create(""));
+	}
+	{
+		Core::AttributeSettings* attribute = RegisterAttribute("IP port", "ipPort", "IP port", Core::ATTRIBUTE_INTERFACETYPE_INTSPINNER);
+		attribute->SetDefaultValue(Core::AttributeInt32::Create(0));
+		attribute->SetMinValue(Core::AttributeInt32::Create(1));
```
Comment (Andrey1994) [inline src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:45]: Min less than default

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp (hunk 2 above)
Comment (Andrey1994) [reply src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:26]: if its possible to implement like IP protocol with names its even better

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	if (auto* currentDevice = GetCurrentDevice())
+	{
+		if (GetBoardID() == currentDevice->GetBoardId() && GetParams() == currentDevice->GetParams())
+			return; // no changes => no need to sync
+
+		// remove current device if there are any changes
+		GetDeviceManager()->RemoveDeviceAsync(currentDevice);
+	}
+
+	CreateNewDevice();
```
Comment (Andrey1994) [inline src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:93]: I think we dont need to do it for each change in attributes, we can create smth like a button to create new device or attribute which will act like a button

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp (hunk 1 above)
Comment (romanov6416) [reply src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:7]: I would add operator== into brainflow

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp (hunk 2 above)
Comment (romanov6416) [reply src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:26]: Currently, I don't see the way how dynamically get board name by board id.
In the beginning, I suggest supporting this feature in brainflow and later in the studio

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp (hunk 1 above)
Comment (Andrey1994) [reply src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:7]: Still need to override == here to take board id into account

File: src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp (hunk 2 above)
Comment (Andrey1994) [reply src/Engine/Devices/BrainFlow/BrainFlowNodes.cpp:26]: Hardcode list like you did for ip protocol, its ok here

**Pull request merged**